### ISSUE TITLE
Some infos about this plugin

### ISSUE THREAD

**Issue by mburger81** [reekoheek/cordova-plugin-xox#1]
Hi can you please explain what does this plugin? We try to set origin header for http request to no be file:// can we do somithing like that wir your plugin?

**Comment by reekoheek** [comment]
Hi, sorry its been a long time for me use this plugin. As I remember, this plugin help you with cross origin xhr by creating other webview with target origin, do xhr there, and send back result to the caller.

**Comment by mburger81** [comment]
Hi, can you help please? Is this plugin you are using or was this only a test or a prof of concept or something else. 
What do you intend with creating other webview? 

Our problem is, we have a ionic angular app und us angulars http and classical html5 websockets.
The problem is cordova and androisa webview set origin to file:// which schema is no allowed on our service we use.
Do you think we can handle this with your plugin or at least could it be a good point to start? 
We googled several days for this problem and asked to millions of people with no valued solution. 
Thx

**Comment by reekoheek** [comment]
Why dont you implement CORS on the API Service and use cordova-plugin-whitelist to achieve that? This plugin, as I remember, I used for development only.

**Comment by mburger81** [comment]
Because we are not the maintainer of the API service, and the service does not support "Origin file://" header. So we have not a problem with CORS because the browser is sending Origin file:// but the API is checking on software side if origin is the right one and return a 403 forbidden.

So CORS and whitelist does not help, right? We have to intercept the request and remove origing header which I think is really not possible, or we have to put on top a revers proxy, which is doubling the quantity of data we have to send.
Server -> Proxy -> AP, Server and Proxy are located on different serve on different locations, because as mentioned the server api is not ours.

**Comment by reekoheek** [comment]
After quick look, I think you dont have to use any plugin to achieve this.

I managed to do XHR from file without CORS with clean cordova init.

What i've done

```bash
cordova create fooApp id.sagara.fooApp fooApp
cd fooApp
cordova platform add android
```

then change the `www/js/index.js` with

```js
var request = new XMLHttpRequest();
request.open('GET', 'http://sagara.id/p/homepage', true);

request.onload = function() {
  if (request.status >= 200 && request.status < 400) {
    // Success!
    console.log('get data text', request.responseText);
  } else {
    // We reached our target server, but it returned an error

  }
};

request.onerror = function() {
    console.error('err', arguments);
  // There was a connection error of some sort
};

request.send();
```

and change `www/index.html` csp section to include the source domain (in this case, **http://sagara.id**)

```html
<meta http-equiv="Content-Security-Policy" content="
            default-src 'self' data: gap: https://ssl.gstatic.com http://sagara.id 'unsafe-eval';
            style-src 'self' 'unsafe-inline';
            media-src *;
            img-src 'self' data: content:;">
```

then run using 

```bash
cordova run android
```

It works!

My system configuration is:

```bash
/tmp/fooApp $ cordova -v
7.0.1

/tmp/fooApp $ cordova plugins
cordova-plugin-whitelist 1.3.2 "Whitelist"
```

On the contrary this plugin actually wants to achieve XHR on plugin development when I want to run js plugin in http server, so it does not have to rebuild on js changes.

**Comment by mburger81** [comment]
Are you sure, sorry I'm not an expert of CSP, but I think CSP in this case doesn't matter for the Origin header?!?
If you do so on the GET which origin do you have in the header?

**Comment by mburger81** [comment]
Try to do a POST instead of a GET passing some parameters and you'll see in the headers you have an Origin: file://

![image](https://user-images.githubusercontent.com/3778892/29563837-997b07c0-873f-11e7-86d5-12743af7b5f3.png)

What I'm trying to do is to omit this origin in the header, and not sure what can help for this the CSP

**Comment by reekoheek** [comment]
Yeah, 403 forbidden is coming from server anyway. recheck the request payload or headers.

I see that the Request Payload is not in the right form, it is [object Object]. Try use query params form or json string, not the object.
